spec/models/program_spec.rb

```diff
@@ -199,4 +200,61 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+    context 'without selected_schedule' do
+      it 'returns false if there are events on the date' do
+        event = create(:event_schedule, start_time: Date.today.to_time)
+        program.selected_schedule.event_schedules = [event]
```

Comment:
there is not selected schedule, so you can't do that. That's why the test is failling